### ISSUE TITLE
Window Resizing issue in Wayland

### ISSUE THREAD

**Issue by raghavdhingra24** [slint-ui/slint#3534]
Hi There,
On Gnome Wayland (Ubuntu 23.04) using Slint 1.2.1 the main window cannot resize to its minimum size although it starts with its minimum size correctly. This issue is not seen on x11 or windows.

Video:
[](url)
[Screencast from 2023-09-23 12-44-03.webm](https://github.com/slint-ui/slint/assets/106472154/0b22ffdd-9f20-4121-a72a-cc1f6908cb56)

Thanks

**Comment by raghavdhingra24** [comment]
Sure, here is the .slint file
```
import {Button, TextEdit, CheckBox, LineEdit, ScrollView, Switch, Slider} from "std-widgets.slint";

component CustomButton inherits Rectangle {
    in-out property text <=> txt.text;
    in-out property <bool> enabled;
    touch := TouchArea {
        clicked => {
            if (root.enabled) {
                root.clicked();
            };
        }
    }
    fc := FocusScope { 
        key-pressed(event) => {
            if (event.text == "\u{0009}") {
                reject
                //self.forward-focus;
            } else if (event.text == "\n" || event.text == " ") {
                touch.clicked();
                accept
            } else {
                accept
            }
        }
    }
    callback clicked();
    enabled: true;
    border-radius: 4px; // Default Value
    border-width: fc.has-focus ? 2px : 1px;
    border-color: fc.has-focus ? rgba(255, 255, 255, 0.12) : rgba(255, 255, 255, 0.1);
    padding-top: 6px;
    padding-bottom: 6px;
    padding-left: 16px;
    padding-right: 16px;
    accessible-role: AccessibleRole.button;
    background: self.enabled ? touch.pressed ? rgba(255, 255, 255, 0.08) : touch.has-hover ? rgba(255, 255, 255, 0.1) : rgba(255, 255, 255, 0.06) : rgba(255, 255, 255, 0.2);
    txt := Text {
        x: (parent.width - self.width)/2;
        y: (parent.height - self.height)/2;
        color: touch.pressed ? #fff : #eee;
        font-size: 1.1rem;
    }
    animate background {
        duration: 100ms;
    }
    height: txt.height + 12px;
}
component CustomLineEdit inherits Rectangle {
    touch := TouchArea { 
    }
    callback clicked <=> touch.clicked;
    in-out property text <=> ti.text;
    in-out property enabled <=> ti.enabled;
    background: ti.enabled ? ti.has-focus ? rgba(255, 255, 255, 0.03) : rgba(255, 255, 255, 0.06) : rgba(255, 255, 255, 0.1);
    border-width: ti.has-focus? 2px : 1px;
    border-color: ti.has-focus ? #fc6c187a : rgba(255, 255, 255, 0.1);
    padding: 8px;
    padding-bottom: 0;
    min-height: self.padding*2 + ti.font-size;
    min-width: 40px;
    animate background {
        duration: 100ms;
    }
    ti := TextInput {
        width: parent.width - parent.padding-left - parent.padding-right - parent.padding;
        height: parent.height;
        color: self.enabled ? white : gray;
        //y: (parent.height - self.height)/2;
        selection-background-color: #ff9355;
        x: max(parent.padding-left, parent.padding);
        font-size: 1.1rem;
        vertical-alignment: TextVerticalAlignment.center;
        wrap: TextWrap.word-wrap;
        input-type: InputType.text;
        enabled: true;
    }
    animate border-width {
        duration: 100ms;
   }
}
export component MainWindow inherits Window {
    background: rgb(10, 10, 10);
    min-height: 100px;
    min-width: 150px;
    padding: 10px;
    title: "Test Slint Program";
    callback button_pressed <=> cb.clicked;
    callback cb_pressed <=> button.clicked;
    in-out property enabled <=> cb.enabled;
    ScrollView  {
        VerticalLayout {
            spacing: 4px;
            width: min(parent.width - 20px, 900px);
            Text {
                text: "hello world";
                color: white;
                horizontal-alignment: center;
                font-size: 1.2rem;
            }
            HorizontalLayout {
                spacing: 0.2rem;
                height: 30px;
                button := Button {
                    preferred-width: parent.width / 2;
                    text: "Button";
                    //min-width: 40px;
                }
                cb := CustomButton { 
                    preferred-width: parent.width / 2;
                    text: "Custom Button";
                    height: parent.height;
                    //enabled: false;
                }
            }
            CheckBox { 
                text: "CheckBox";
            }
            HorizontalLayout {
                spacing: 0.2rem;
                LineEdit { 
                    preferred-width: parent.width / 2;
                    text: "LineEdit";
                }
                CustomLineEdit {
                    preferred-width: parent.width / 2;
                    text: "CustomTextBox";
                    border-radius: 0.4rem;
                }
            }
            Switch { 
                text: "Switch";
            }
            TextInput {
                text: "TextInput";
                color: white;
            }
            Slider { 
                width: parent.width;
            }
        }
    }
}

```

**Comment by ogoffart** [comment]
I cannot reproduce it with Weston. I will have to try with Gnome.

**Comment by raghavdhingra24** [comment]
I had checked on KDE Plasma 5.27's wayland session, the issue was not present there. Seems to be only present in gnome wayland

**Comment by ogoffart** [comment]
Could be fixed with #4032 , please try again.

**Comment by raghavdhingra24** [comment]
Yes, the bug is fixed, now the window is resizing properly
Thanks
